components/plate_based_widgets/mapeditor/PlateMapWidget.vue

```diff
@@ -0,0 +1,737 @@
+        @click-shift-exact="basic_shift_or_ctrl_select(well_index)"
+      />
+    </span>
+  </div>
+</template>
+<script>
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faMinusCircle, faPlusCircle } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import PlateWell from "@/components/basic_widgets/PlateWell.vue";
+
+library.add(faMinusCircle);
+library.add(faPlusCircle);
+
+const no_stroke_width = 0;
+const hover_stroke_width = 2;
+const selected_stroke_width = 4;
+const hover_color = "#ececed";
+const selected_color = "#FFFFFF";
+const default_color = "#b7b7b7";
+const debug_mode = undefined;
+
+export default {
+  name: "PlateMap",
+  components: { FontAwesomeIcon, PlateWell },
+  props: {
+    selected: {
+      type: Array,
+      default: function () {
+        return new Array(24).fill(false);
+      },
+    },
+    platecolor: {
+      type: Array,
+      default: function () {
+        return new Array(24).fill(default_color);
+      },
+    },
+  },
+  data() {
+    return {
+      all_select_or_cancel: true,
+      x_origin: 0,
+      y_origin: 0,
+      rect_width: 0,
+      rect_height: 0,
+      all_select: this.selected,
+      hover: new Array(24).fill(false),
+      hover_color: new Array(24).fill(hover_color),
+      stroke_width: new Array(24).fill(no_stroke_width),
+      row_values: ["A", "B", "C", "D"],
+      column_values: ["1", "2", "3", "4", "5", "6"],
+      testerf: false,
+    };
+  },
+  watch: {
+    selected: function () {
+      // reset stroke width when selected wells is reset
+
+      if (this.selected.filter((x) => x).length === 0) {
+        this.all_select = Array(24).fill(false);
+        this.stroke_width = Array(24).fill(no_stroke_width);
+      }
+    },
+  },
+  created() {
+    this.stroke_width.splice(0, this.stroke_width.length);
+    for (let j = 0; j < this.all_select.length; j++) {
+      this.stroke_width[j] = !this.all_select[j] ? no_stroke_width : selected_stroke_width;
+    }
+    const allEqual = (arr) => arr.every((v) => v === true); // verify in the pre-select all via a const allEqual function.
+    this.all_select_or_cancel = allEqual(this.all_select) ? false : true; // if pre-select has all wells is true, then toggle from (+) to (-) icon.
+  },
+  methods: {
+    column_left_offset(column) {
+      switch (column) {
+        case "1":
+          return "35.9792";
+        case "2":
+          return "97.5836";
+        case "3":
+          return "159.188";
+        case "4":
+          return "220.792";
+        case "5":
+          return "282.397";
+        case "6":
+          return "344.001";
+      }
+    },
+    row_top_offset(row) {
+      switch (row) {
+        case "A":
+          return "41.928";
+        case "B":
+          return "103.621";
+        case "C":
+          return "165.779";
+        case "D":
+          return "224.1";
+      }
+    },
+    well_top_left_offset(index) {
```

Comment:
I'm sure there's a way to make this function use `column_left_offset` and `row_top_offset`. You can get the row name with something like `String.fromCharCode(index % 4 + 65)` and the column with `index // 6`. If not then at least using `index % 4 + 65` to get the `top` value and `index // 6` to get the `left` value would still be better